### PR TITLE
feat(remote): add SSH remote control plane MVP with chat + rollout safeguards

### PR DESCRIPTION
**Pull request by Keith-CY**
## Summary
- add Carrier remote control plane APIs for SSH-managed hosts/instances (no remote Carrier dependency)
- add provider profile CRUD/binding flows and remote profile test host selection
- add unified local/remote chat stream support in WebUI
- add rollout safeguards and observability gate signals (`healthy|canary|hold`)
- migrate WebUI source workflow to TypeScript build pipeline (Bun/esbuild -> static JS assets)

## Key Changes
- Gateway:
  - new remote host/instance/config/session/memory/profile/binding/chat APIs
  - feature flags endpoint and dependency-normalized effective flags
  - remote metrics collector with rollout gate output
- WebUI:
  - new routes: `#/servers`, `#/profiles`, `#/remote-chat`, `#/remote-observability`
  - edit/cancel/update closures for hosts and profiles
  - binding delete UX and explicit profile test host selector
  - observability rollout card + status line
- Tooling/Release:
  - add `scripts/build-webui.sh`
  - include webui build in Makefile/release workflow
- Docs/Runbooks:
  - update go-live/post-merge smoke checklists and README env/metrics notes

## Verification
- `go test ./...` (gateway)
- `bash scripts/build-webui.sh`
- `bun run test tests/remote-control-plane.spec.ts tests/remote-observability.spec.ts` (webui/e2e)

### CODE REVIEW

File: gateway/remote_openclaw.go [hunk 1]
```diff
@@ -0,0 +1,669 @@
...
+
+func remoteArchiveSession(ctx context.Context, host RemoteHost, agentID, sessionID string) ([]remoteExecResult, error) {
+	if err := validateAgentIdentifier(agentID); err != nil {
+		return nil, err
+	}
+	trimmedSession := strings.TrimSpace(sessionID)
+	if trimmedSession == "" {
+		return nil, fmt.Errorf("sessionId is required")
+	}
+	cmd := fmt.Sprintf("set -e; base=\"$HOME/.openclaw/agents/%s\"; src=\"$base/sessions/%s.jsonl\"; dst=\"$base/sessions_archive/%s.jsonl\"; if [ -f \"$src\" ]; then mkdir -p \"$base/sessions_archive\"; mv \"$src\" \"$dst\"; exit 0; fi; if [ -f \"$dst\" ]; then exit 0; fi; exit 44", agentID, trimmedSession, trimmedSession)
```
Comment (Akane-CN) [inline gateway/remote_openclaw.go]: BS: `sessionID` is interpolated directly into shell command strings here (and similarly in `remoteDeleteSession` below), allowing shell injection via crafted session IDs from the API path. Please validate `sessionID` against a strict allow-list (or quote safely + reject unsafe chars) before shell construction.

**Comment by Keith-CY** [comment]
Addressed the blocking review findings in commit `71aea3f`.

### Fixed
1. **Session command injection risk (BS)**
- Added strict `sessionId` allow-list validation (`validateRemoteSessionIdentifier`) and enforced it in API + remote session ops.
- Session archive/delete shell commands now pass session id via shell-quoted variable (`sid=...`) instead of direct interpolation.
- Invalid `sessionId` now returns `400 E_USAGE` before any SSH execution.

2. **Snapshot path expansion bug (BS)**
- `remotePatchConfig` snapshot path now uses shell-expanded path assignment under `~/.openclaw/snapshots/`.
- Removed single-quoted literal `$HOME...` target that prevented shell expansion.

### Extra robustness updates
- `streamChatSSE` now checks `writeSSEEvent` errors and exits early on write failure.
- `chunkString` now chunks on rune boundaries (avoids UTF-8 split issues).
- Added README note about SSH TOFU behavior (`StrictHostKeyChecking=accept-new`).

### Tests added
- `TestRemoteSessionActionsRejectInvalidSessionID`
- `TestRemotePatchConfigUsesExpandedSnapshotPath`

### Verification
- `go test ./...` (gateway)
- `bun run test tests/remote-control-plane.spec.ts tests/remote-observability.spec.ts`

File: gateway/remote_openclaw.go [hunk 2]
```diff
@@ -0,0 +1,669 @@
...
+		end = lastArray
+	}
+	if end >= 0 {
+		maybe := strings.TrimSpace(candidate[:end+1])
+		if json.Valid([]byte(maybe)) {
+			return maybe
+		}
+	}
+	return ""
+}
```
[the remark this answers is not in the document]
Comment (Keith-CY) [reply gateway/remote_openclaw.go]: Thanks — agreed this is a valid robustness concern.\n\nFor this MVP PR, we are deferring this to follow-up because current OpenClaw CLI output is not yet guaranteed as strict `json-only` across all runtime paths. The current extractor keeps compatibility with mixed stdout/stderr output and non-JSON fallback behavior.\n\nFollow-up plan:\n1. align with OpenClaw CLI on a stable machine-readable output contract (`--json-only` or equivalent),\n2. switch remote commands to that contract,\n3. simplify/remove heuristic extraction in `extractJSONObjectOrArray`,\n4. add targeted regression tests for mixed-output edge cases.\n\nMarking this thread resolved for MVP scope tracking.
